packages/labs/test/selectTests.tsx

```diff
@@ -98,39 +98,127 @@ describe("<Select>", () => {
+        const wrapper = select({ onQueryChange, resetOnClose: true, popoverProps: {}, query: "1972" });
+        wrapper.find("table").simulate("click");
+        assert.isTrue(wrapper.find(Popover).prop("isOpen"));
+        wrapper.find("table").simulate("click");
+        assert.isFalse(wrapper.find(Popover).prop("isOpen"));
+        assert.isTrue(onQueryChange.calledOnce);
+        assert.isTrue(onQueryChange.calledWithExactly(""));
+    });
+
+    it("onQueryChange is called when clear button pressed", () => {
+        const onQueryChange = sinon.spy();
+        const wrapper = select({ onQueryChange, query: "1972" });
+        wrapper
+            .find(InputGroup)
+            .find(Button)
+            .simulate("click");
+        assert.isTrue(onQueryChange.calledOnce);
+        assert.isTrue(onQueryChange.calledWithExactly(""));
+    });
+
     it("input can be controlled with inputProps", () => {
         const value = "nailed it";
         const onChange = sinon.spy();
 
-        const input = select({ inputProps: { value, onChange } }).find("input");
+        const wrapper = select({ inputProps: { value, onChange } });
+        assert.equal(wrapper.state("query"), value);
+
+        const input = wrapper.find("input");
         assert.equal(input.prop("value"), value);
+        assert.equal(wrapper.state("query"), value);
 
         input.simulate("change");
         assert.isTrue(onChange.calledOnce);
+
+        const value2 = "nailed it again";
+        wrapper.setProps({ inputProps: { value: value2 } });
+        assert.equal(wrapper.find("input").prop("value"), value2);
+        assert.equal(wrapper.state("query"), value2);
     });
 
     it("popover can be controlled with popoverProps", () => {
         // Select defines its own popoverWillOpen so this ensures that the passthrough happens
         const popoverWillOpen = sinon.spy();
         const tetherOptions = {}; // our own instance
-        const wrapper = select({ popoverProps: { isOpen: undefined, popoverWillOpen, tetherOptions } });
+        const wrapper = select({ popoverProps: { popoverWillOpen, tetherOptions } });
         wrapper.find("table").simulate("click");
         assert.strictEqual(wrapper.find(Popover).prop("tetherOptions"), tetherOptions);
         assert.isTrue(popoverWillOpen.calledOnce);
     });
 
+    it("popover can be controlled with popoverProps.isOpen", () => {
+        const wrapper = select({ popoverProps: { isOpen: false } });
+        wrapper.find("table").simulate("click");
+        assert.strictEqual(wrapper.find(Popover).prop("isOpen"), false);
+        wrapper.setProps({ popoverProps: { isOpen: true } });
+        assert.strictEqual(wrapper.find(Popover).prop("isOpen"), true);
+    });
+
     it("returns focus to focusable target after popover closed");
 
-    function select(props: Partial<ISelectProps<Film>> = {}, query?: string) {
-        const wrapper = mount(
+    function select(props: Partial<ISelectProps<Film>> = {}) {
```

Comment:
Per my comment above, would prefer to undo changes to this function to let uncontrolled mode remain testable.